src/Maestro/Maestro.Data/BuildAssetRegistryContext.cs

```diff
@@ -352,6 +352,17 @@ where buildIds.Contains(build.Id)
             }
             return dict.Values.ToList();
         }
+
+        public bool IsProductDependency(int buildId, string repositoryName, string branchName)
+        {
+            return BuildDependencies
+                .Any(d => d.IsProduct
```

Comment:
What will happen if a dependency used to be a product dependency, but is now a tooling dependency? Would that still be marked as a product dependency?

Reply:
Well, I guess if it's the same dependent build id that probably wouldn't happen. But just something to consider.

Reply:
So as I understand there's no changes needed here?

Reply:
I think this logic is sound, it looks good to me. As always, we're trying to predict future behavior with past behavior here. So for instance, if I initially apply a subscription to foorepo@main, and build 123 was product, and then I decide it should be toolset, I go in and change that. But until that new commit builds and reports to BAR, this code will still report this as a product. dependency.

I think this is fine.